WordPress/gutenberg
# MetaBoxes: Moving meta boxes out of the generic editor module

The `editor` module is a generic module meant to be used to build any editing interface (template editor, post editor, p2 editor...) it shouldn't be concerned about the meta boxes which are post editor specific.

I'm pretty happy with this PR, I was able to extract all the meta box logic out of the editor module by doing:

 - Moving the components to the edit-post module
 - Moving the metabox state to the edit-post module
 - Hooking into the `PostSavedStatus` component to show the `isLoading` state when the meta boxes are loading and to always consider the post dirty if there are meta boxes on the page
 - Triggering meta boxes saving by watching `isSavingPost` selector. When the post was being saved and it's not anymore, trigger the meta boxes save behavior.

**Testing instructions**

 - Add some metaboxes
 - The "save" link should always be visible
 - When you save a post, the meta boxes are saved as well
 - When the meta boxes are saving, the top bar shows a loading state.
